api-client/src/main/java/com/desk/java/apiclient/DeskClient.java

```diff
@@ -158,339 +101,157 @@ public String getUrl(String path) {
-    }
-
-    /**
-     * Package access for testing purposes
-     */
-    RestAdapter getRestAdapter() {
-        return restAdapter;
-    }
-
-    private RestAdapter createRestAdapter() {
-        RestAdapter.Builder builder = new RestAdapter.Builder()
-                .setEndpoint(getUrl(API_BASE_PATH))
-                .setClient(createOkClient())
-                .setConverter(new GsonConverter(createGson()));
-
-        // if debug is set turn on logging
-        if (isDebug) {
-            builder.setLogLevel(RestAdapter.LogLevel.FULL);
-        }
-
-        // add the request interceptor of we have one
-        if (requestInterceptor != null) {
-            builder.setRequestInterceptor(requestInterceptor);
-        }
-
-        return builder.build();
-    }
-
-    private OkClient createOkClient() {
-        switch (authType) {
-
-            case OAUTH:
-                if (oAuthConsumer == null) {
-                    throw new IllegalStateException("a RetrofitHttpOAuthConsumer must be created before creating OKClient");
-                }
-                return new OAuthSigningOkClient(createHttpClient(), oAuthConsumer);
-            case API_TOKEN:
-                return new ApiTokenSigningOkClient(createHttpClient(), apiToken);
-            default:
-                throw new IllegalStateException("AuthType " + authType + " isn't supported.");
-        }
-    }
-
-    private Gson createGson() {
-        return new GsonBuilder()
-                .registerTypeAdapter(Date.class, new ISO8601DateAdapter())
-                .registerTypeAdapter(CaseLock.class, CaseLock.TYPE_ADAPTER)
-                .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
-                .create();
-    }
-
-    private OkHttpClient createHttpClient() {
-        OkHttpClient httpClient = new OkHttpClient();
-
-        // if we have response cache let's use it!
-        if (responseCache != null) {
-            httpClient.setCache(responseCache);
-        }
-
-        return httpClient;
-    }
-
-    private RetrofitHttpOAuthConsumer createOAuthConsumer() {
-        if (OAUTH == authType) {
-            RetrofitHttpOAuthConsumer consumer = new RetrofitHttpOAuthConsumer(consumerKey, consumerSecret);
-            consumer.setTokenWithSecret(accessToken, accessTokenSecret);
-            return consumer;
-        } else {
-            return null;
-        }
-    }
+    InboundMailboxService inboundMailboxes();
 
     /**
      * Builder class to assist in create a desk client instance
      */
-    public static class Builder {
-
-        private Cache responseCache;
-        private RequestInterceptor requestInterceptor;
-        private String hostname;
-        private String apiToken;
-        private String consumerKey;
-        private String consumerSecret;
-        private String accessToken;
-        private String accessTokenSecret;
-        private AuthType authType;
-        private boolean isDebug;
+    class Builder {
```

Comment:
I feel like this `Builder` should belong in `DefaultDeskClient` since it creates a `DefaultDeskClient`. In it's current state you can call `new RxDeskClient.Builder("hostname", "apiToken").create()` thinking it would create a `RxDeskClient` but it actually creates a `DefaultDeskClient`.